packages/toolpad-app/src/ThemeContext.tsx

```diff
@@ -0,0 +1,75 @@
+import * as React from 'react';
+import { ThemeProvider as MuiThemeProvider, createTheme } from '@mui/material/styles';
+import { deepmerge } from '@mui/utils';
+import useMediaQuery from '@mui/material/useMediaQuery';
+import { unstable_useEnhancedEffect as useEnhancedEffect } from '@mui/material/utils';
+import { getDesignTokens, getThemedComponents, getMetaThemeColor } from './theme';
+
+interface ThemeProviderProps {
+  children?: React.ReactNode;
+}
+
+export type PaletteModeOptions = 'light' | 'dark';
+
+export const DispatchContext = React.createContext<
+  React.Dispatch<React.SetStateAction<PaletteModeOptions>>
+>(() => {
+  throw new Error('Forgot to wrap component in `ThemeProvider`');
+});
+
+export function ThemeProvider(props: ThemeProviderProps) {
+  const { children } = props;
+  const prefersDarkMode = useMediaQuery('(prefers-color-scheme: dark)');
+  const preferredMode = prefersDarkMode ? 'dark' : 'light';
+  const [paletteMode, setPaletteMode] = React.useState<PaletteModeOptions>(preferredMode);
+
+  React.useEffect(() => {
+    if (typeof window !== 'undefined') {
+      setPaletteMode(
+        JSON.parse(localStorage.getItem('paletteMode') as PaletteModeOptions) || preferredMode,
+      );
+    }
+  }, [preferredMode]);
+
+  React.useEffect(() => {
+    const metas = document.querySelectorAll('meta[name="theme-color"]');
+    metas.forEach((meta) => {
+      meta.setAttribute('content', getMetaThemeColor(paletteMode));
```

Comment:
Can [`next/head`](https://nextjs.org/docs/api-reference/next/head) be used for this purpose?

Reply:
Do you suggest something like importing a `useTheme` hook into `ToolpadShell` and using 

```tsx
<Head> 
   <meta
            name="theme-color"
            content={getMetaThemeColor(theme.palette.mode)}
            media=`(prefers-color-scheme: ${theme.palette.mode})`
          />
</Head>
 ```

Reply:
I believe it can just go here in `Themeprovider`, and we can then remove it from `_document.ts`